pydantic/tools.py

```diff
@@ -0,0 +1,48 @@
+from functools import lru_cache
+from pathlib import Path
+from typing import Any, Callable, Type, TypeVar, Union
+
+from pydantic.parse import Protocol, load_file
+
+from .typing import display_as_type
+
+__all__ = ('parse_obj',)
+
+NameGenerator = Callable[[Type[Any]], str]
+
+
+def _generate_parsing_type_name(type_: Any) -> str:
+    return f'ParsingModel[{display_as_type(type_)}]'
+
+
+@lru_cache(maxsize=2048)
+def _get_parsing_type(type_: Any, type_name: Union[str, NameGenerator] = None) -> Any:
+    from pydantic.main import create_model
+
+    if type_name is None:
+        type_name = _generate_parsing_type_name
+    if not isinstance(type_name, str):
+        type_name = type_name(type_)
+    return create_model(type_name, obj=(type_, ...))
+
+
+T = TypeVar('T')
+
+
+def parse_obj(type_: Type[T], obj: Any, type_name: Union[str, NameGenerator] = None) -> T:
```

Comment:
I went ahead and renamed this to `parse_obj` for consistency with #481 and the `BaseModel` method; let me know if I should change it back.